File: containers/dotnet.py
from .base import Centos_7


class Dotnet( Centos_7 ):
    client_branch = 'main'
    client_service = 'sigrha_clients.service'
    client_service_path = 'dotnet/sigrha_clients.service'
    client_service_env = 'dotnet/sigrha_clients_test.env'

    gateway_branch = 'main'
    gateway_service = 'sigrha_gateway.service'
    gateway_service_path = 'dotnet/sigrha_gateway.service'
    gateway_service_env = 'dotnet/sigrha_gateway.env'

    opportunities_branch = 'main'
    opportunities_service_path = 'dotnet/sigrha_opportunities.service'
    opportunities_service = 'sigrha_opportunities.service'
    opportunities_env = 'sigrha_opportunities.env'

    migration_mode = 'dotnet/database_migration.sh'
    build_mode = 'dotnet/build_project.sh'

    scripts = (
        'ssh/provision.py',
        ( "systemd/cp.py", 'cls.client_service_path' ),
        ( "systemd/cp.py", 'cls.opportunities_service_path' ),
        ( "systemd/cp.py", 'cls.gateway_service_path' ),

        'dotnet/install.py',
        'dotnet/post_install.sh',
        (
            'git_clone.py',
            'git@github.com:AptudeSiGRHA/clients_service.git',
            'cls.client_branch', 'main' ),
        (
            'git_clone.py',
            'git@github.com:AptudeSiGRHA/opportunities_service.git',
            'cls.opportunities_branch', 'main' ),
        (
            'git_clone.py',
            'git@github.com:AptudeSiGRHA/Gateway.git',
            'cls.gateway_branch', 'main' ),
        (
            'cls.migration_mode',
            '/home/chibi/projects/clients_service__main/API_Clients/',
            'cls.client_service_env',
        ),
        (
            'cls.migration_mode',
            '/home/chibi/projects/opportunities_service__main/Opportunities/',
            'cls.opportunities_env',
        ),
        (
            'cls.build_mode',
            '/home/chibi/projects/Gateway__main/Gateway/Gateway/',
            'cls.gateway_service_env',
        ),
        ( "systemd/systemd.py", 'enable', 'cls.client_service' ),
        ( "systemd/systemd.py", 'restart', 'cls.client_service' ),

        ( "systemd/systemd.py", 'enable', 'cls.opportunities_service' ),
        ( "systemd/systemd.py", 'restart', 'cls.opportunities_service' ),

        ( "systemd/systemd.py", 'enable', 'cls.gateway_service' ),
        ( "systemd/systemd.py", 'restart', 'cls.gateway_service' ),
    )
    env_vars = {
        'HOME': '/root/'
    }


class Mitsuha( Dotnet ):
    pass


class Kaoru( Dotnet ):
    client_branch = 'development'
    opportunities_branch = 'development'
    gateway_branch = 'development'

    client_service = 'sigrha_clients_test.service'
    client_service_path = 'dotnet/sigrha_clients_test.service'
    client_service_env = 'dotnet/sigrha_clients_test.env'

    gateway_service = 'sigrha_gateway.service_test'
    gateway_service_path = 'dotnet/sigrha_gateway_test.service'
    gateway_service_env = 'dotnet/sigrha_gateway_test.env'

    opportunities_service_path = 'dotnet/sigrha_opportunities_test.service'
    opportunities_service = 'sigrha_opportunities_test.service'
    opportunities_env = 'sigrha_opportunities_test.env'
    migration_mode = 'dotnet/database_migration_debug.sh'
    build_mode = 'dotnet/build_project_debug.sh'

File: weeaboo/machines/dotnet.rb
Dir[ File.dirname(__FILE__) + "/base.rb" ].each { |file| require file }

class Dot_net < Base_centos
	def initialize( params={} )
		super()
		@abstract = true
		@name = 'Dot Net'
		@box = "base_centos_7"
		@ram = 1000
		@scripts = [
			Script.new( "provision/update_python_lib.sh" ),
			Python.new( "provision/add_user.py",
				args: [ 'chibi' ] ),

			Python.new( "provision/copy_host.py" ),
			Python.new( "provision/repos/cp_all_repos.py" ),
			Python.new( "provision/dotnet/install.py" ),
			Script.new( "provision/dotnet/post_install.sh" ),
			Python.new( "provision/ssh/provision.py" ),

			Python.new( "provision/systemd/cp.py",
				args: [ "dotnet/#{params.fetch( :service_client )}.service" ] ),
			Python.new( "provision/systemd/cp.py",
				args: [ "dotnet/#{params.fetch( :service_oportunities )}.service" ] ),
			Python.new( "provision/systemd/cp.py",
				args: [ "dotnet/#{params.fetch( :service_gateway )}.service" ] ),

			Python.new( "provision/git_clone.py",
				args: [
					params.fetch(
						:client_repo,
						'git@github.com:AptudeSiGRHA/clients_service.git', ),
					params.fetch( :client_branch, 'main', ),
					params.fetch( :client_repo_name, 'main', ),
				] ),
			Python.new( "provision/git_clone.py",
				args: [
					params.fetch(
						:opportunities_repo,
						'git@github.com:AptudeSiGRHA/opportunities_service.git', ),
					params.fetch( :opportunities_branch, 'main', ),
					params.fetch( :opportunities_repo_name, 'main', ),
				] ),
			Python.new( "provision/git_clone.py",
				args: [
					params.fetch(
						:login_repo,
						'git@github.com:AptudeSiGRHA/ADLoginService.git', ),
					params.fetch( :login_branch, 'main', ),
					params.fetch( :login_repo_name, 'main', ),
				] ),
			Python.new( "provision/git_clone.py",
				args: [
					params.fetch(
						:gateway_repo,
						'git@github.com:AptudeSiGRHA/Gateway.git', ),
					params.fetch( :gateway_branch, 'main', ),
					params.fetch( :gateway_repo_name, 'main', ),
				] ),

			Script.new( "provision/dotnet/database_migration.sh",
				args: [
					'/home/chibi/projects/clients_service__main/API_Clients/',
					"/etc/systemd/system/#{params.fetch( :service_client )}.env",
				] ),
			Script.new( "provision/dotnet/database_migration.sh",
				args: [
					'/home/chibi/projects/opportunities_service__main/Opportunities/',
					"/etc/systemd/system/#{params.fetch( :service_oportunities )}.env",
				] ),
			Script.new( "provision/dotnet/build_project.sh",
				args: [
					'/home/chibi/projects/Gateway__main/Gateway/Gateway/',
				] ),

			Python.new( "provision/systemd/systemd.py",
				args: [ 'enable', "#{params.fetch( :service_client )}.service" ] ),
			Python.new( "provision/systemd/systemd.py",
				args: [ 'start', "#{params.fetch( :service_client )}.service" ] ),
			Python.new( "provision/systemd/systemd.py",
				args: [ 'enable', "#{params.fetch( :service_oportunities )}.service" ] ),
			Python.new( "provision/systemd/systemd.py",
				args: [ 'start', "#{params.fetch( :service_oportunities )}.service" ] ),
			Python.new( "provision/systemd/systemd.py",
				args: [ 'enable', "#{params.fetch( :service_gateway )}.service" ] ),
			Python.new( "provision/systemd/systemd.py",
				args: [ 'start', "#{params.fetch( :service_gateway )}.service" ] ),
		]
	end
end

class Mitsuha < Dot_net
	def initialize()
		super(
			:service_gateway => 'sigrha_gateway',
			:service_client => 'sigrha_clients',
			:service_oportunities => 'sigrha_opportunities',
		)
		@abstract = false
		@name = 'Mitsuha'
	end
end

class Kaoru < Dot_net
	def initialize()
		super(
			:client_branch => 'development',
			:opportunities_branch => 'development',
			:gateway_branch => 'development',
			:login_branch => 'main',

			:service_gateway => 'sigrha_gateway_test',
			:service_client => 'sigrha_clients_test',
			:service_oportunities => 'sigrha_opportunities_test',
		)
		@abstract = false
		@name = 'Kaoru'
	end
end
